bindings/distrdf/python/DistRDF/Ranges.py

```diff
@@ -238,30 +248,69 @@ def get_percentage_ranges(treenames: List[str], filenames: List[str], npartition
     if friendinfo is not None:
         # We need to transmit the full list of treenames and filenames to each
         # task, in order to properly align the full dataset considering friends.
-        return [
+        if sampleMap is not None:
+                
+            samples = []
+
+            for filename, treename in zip(filenames, treenames):
+                sample = sampleMap.get(filename + "/" + treename)
+                samples.append(sample)
+
+            return [
             TreeRangePerc(
                 exec_id, rangeid, treenames, filenames, start_sample_idxs[rangeid], end_sample_idxs[rangeid],
-                first_tree_start_perc_tasks[rangeid], last_tree_end_perc_tasks[rangeid], friendinfo)
+                first_tree_start_perc_tasks[rangeid], last_tree_end_perc_tasks[rangeid], friendinfo, samples)
             for rangeid in range(npartitions)
-        ]
+            ]
+
+        else:   
+            return [
+                TreeRangePerc(
+                    exec_id, rangeid, treenames, filenames, start_sample_idxs[rangeid], end_sample_idxs[rangeid],
+                    first_tree_start_perc_tasks[rangeid], last_tree_end_perc_tasks[rangeid], friendinfo, [])
```

Comment:
This would be more semantically correct
```suggestion
                    first_tree_start_perc_tasks[rangeid], last_tree_end_perc_tasks[rangeid], friendinfo, None)
```